aea/multiplexer.py

```diff
@@ -253,6 +276,10 @@ async def _stop(self) -> None:
             if c.state in (ConnectionStates.connecting, ConnectionStates.connected)
         ]:
             await connection.disconnect()
+
```

Comment:
I am confused how this related to `disconnect` above. Should we not be calling `disconnect` from here and implement the disconnection in `disconnect`?

Reply:
updated.